### DISCUSSION TITLE
So-firewall issue

### DISCUSSION THREAD

**Discussion by OlexTratisky** [Security-Onion-Solutions/securityonion#9136] [General]
Hello,

I am trying to allow a windows-client running wazuh-agent to communicate with SO wazuh server. Tried to add the windows client ip using so-firewall to all the wazuh-* services and did a so-allow apply. Still the windows machine gives connection refused when trying to manually connect over port 1514 or 1515 over TCP and UDP.

What i tried next is to manuall play around with so-firewall here is what i did: 1) added a new  hostgroup called wazuh. a

so-firewall listhostgroups
analyst
beats_endpoint
beats_endpoint_ssl
elasticsearch_rest
endgame
fleet
heavy_node
idh
manager
minion
node
osquery_endpoint
receiver
search_node
sensor
strelka_frontend
syslog
wazuh
wazuh_agent
wazuh_api
wazuh_authd

so-firewall includedhosts wazuh
10.30.30.21

so-firewall listportgroups
wazuh

vi /opt/so/saltstack/local/pillar/minions/securityonion_standalone.sls and appended to the last lines the following section for binding:
  78 firewall:
     79   assigned_hostgroups:
     80     chain:
     81       DOCKER-USER:
     82         hostgroups:
     83          wazuh:
     84             portgroups:
     85               - portgroups.wazuh

  salt securityonion_standalone state.apply firewall

All succeeds but the windows machine stil lcannot communicate with the machine over port 1514 and 1515. 

Please note that the web administration console works fine from that machine and can open the SO on port 443.

What am i missing here?

--------
          ID: insert_DOCKER-USER_wazuh_api_10.30.30.21/32_55000_tcp_79
    Function: iptables.insert
      Result: True
     Comment: iptables rule for insert_DOCKER-USER_wazuh_api_10.30.30.21/32_55000_tcp_79 already set for ipv4 (/usr/sbin/iptables --wait -t filter -I DOCKER-USER 1 -p tcp --dport 5
5000 --source 10.30.30.21/32 --jump ACCEPT)
     Started: 09:46:48.871667
    Duration: 21.595 ms
     Changes:
----------
          ID: insert_DOCKER-USER_wazuh_authd_10.30.30.21/32_1515_tcp_80
    Function: iptables.insert
      Result: True
     Comment: iptables rule for insert_DOCKER-USER_wazuh_authd_10.30.30.21/32_1515_tcp_80 already set for ipv4 (/usr/sbin/iptables --wait -t filter -I DOCKER-USER 1 -p tcp --dport
1515 --source 10.30.30.21/32 --jump ACCEPT)
     Started: 09:46:48.893672
    Duration: 21.551 ms
     Changes:
----------
          ID: insert_DOCKER-USER_wazuh_authd_10.30.0.0/16_1515_tcp_81
    Function: iptables.insert
      Result: True
     Comment: iptables rule for insert_DOCKER-USER_wazuh_authd_10.30.0.0/16_1515_tcp_81 already set for ipv4 (/usr/sbin/iptables --wait -t filter -I DOCKER-USER 1 -p tcp --dport 151
5 --source 10.30.0.0/16 --jump ACCEPT)
     Started: 09:46:48.915613
    Duration: 22.168 ms
     Changes:
----------
          ID: insert_DOCKER-USER_wazuh_authd_10.30.166.21_1515_tcp_82
    Function: iptables.insert
      Result: True
     Comment: iptables rule for insert_DOCKER-USER_wazuh_authd_10.30.30.21_1515_tcp_82 already set for ipv4 (/usr/sbin/iptables --wait -t filter -I DOCKER-USER 1 -p tcp --dport 151
5 --source 10.30.30.21 --jump ACCEPT)
     Started: 09:46:48.938180
    Duration: 21.845 ms
     Changes:

Thanks

**Comment by dougburks** [comment]
Are you trying to connect the Wazuh agent to the Security Onion manager? If so, have you tried a simple `so-allow`?
https://docs.securityonion.net/en/2.3/so-allow.html

**Comment by OlexTratisky** [comment]
Hello, yes i did try that as a first attempt. It did not work i am re-installing SO since i believe the iptables rules are kind of messed up now

Thanks again.
